cpu/native/startup.c

```diff
@@ -74,73 +99,48 @@ void _native_null_in(char *stdiotype)
 }
 
 /**
- * set up stdout redirection
+ * @brief   set up output redirection
  *
- * @param stdouttype: "stdio" (leave intact), "null" (redirect to
- * /dev/null) or "file" (redirect to /tmp/riot.stdout.PID)
- */
-void _native_log_stdout(char *stdouttype)
-{
-    int stdout_outfile;
-
-    if (strcmp(stdouttype, "stdio") == 0) {
-        _native_null_out_file = -1;
-        return;
-    }
-    else if (strcmp(stdouttype, "null") == 0) {
-        if ((stdout_outfile = real_open("/dev/null", O_WRONLY)) == -1) {
-            err(EXIT_FAILURE, "_native_log_stdout: open");
-        }
-    }
-    else if (strcmp(stdouttype, "file") == 0) {
-        char stdout_logname[255];
-        snprintf(stdout_logname, sizeof(stdout_logname), "/tmp/riot.stdout.%d", _native_pid);
-        if ((stdout_outfile = real_creat(stdout_logname, 0666)) == -1) {
-            err(EXIT_FAILURE, "_native_log_stdout: open");
-        }
-    }
-    else {
-        errx(EXIT_FAILURE, "_native_log_stdout: unknown log type");
-    }
-
-    if (real_dup2(stdout_outfile, STDOUT_FILENO) == -1) {
-        err(EXIT_FAILURE, "_native_log_stdout: dup2(STDOUT_FILENO)");
-    }
-    _native_null_out_file = stdout_outfile;
-}
-
-/**
- * set up stderr redirection
+ * @param[in] stdiotype The type of redirection
+ * @param[in] output    Output file. May be either `STDOUT_FILENO` for stdout or
+ *                      `STDERR_FILENO` for stderr
  *
- * @param stderrtype: "stdio" (leave intact), "null" (redirect to
- * /dev/null) or "file" (redirect to /tmp/riot.stdout.PID)
+ * @return The new file descriptor of the redirection
+ * @return -1 if the file descriptor did not change from the standard one
  */
-void _native_log_stderr(char *stderrtype)
+int _native_log_output(_stdiotype_t stdiotype, int output)
 {
-    int stderr_outfile;
+    int outfile;
 
-    if (strcmp(stderrtype, "stdio") == 0) {
-        return;
-    }
-    else if (strcmp(stderrtype, "null") == 0) {
-        if ((stderr_outfile = real_open("/dev/null", O_WRONLY)) == -1) {
-            err(EXIT_FAILURE, "_native_log_stderr: open");
-        }
-    }
-    else if (strcmp(stderrtype, "file") == 0) {
-        char stderr_logname[255];
-        snprintf(stderr_logname, sizeof(stderr_logname), "/tmp/riot.stderr.%d", _native_pid);
-        if ((stderr_outfile = real_creat(stderr_logname, 0666)) == -1) {
-            err(EXIT_FAILURE, "_native_log_stderr: open");
+    assert((output == STDERR_FILENO) || (output == STDOUT_FILENO));
+
+    switch (stdiotype) {
+        case _STDIOTYPE_STDIO:
+            return -1;
```

Comment:
Remember: we are talking about the error-handling of some file-internal function that executes commands by a value that was set due to user input. ;-)